sn_networking/src/record_store.rs

```diff
@@ -448,6 +448,48 @@ impl NodeRecordStore {
 
         Ok(())
     }
+
+    // When the accumulated record copies exceeds the `expotional pricing point` (max_records * 0.6)
+    // those `out of range` records shall be cleaned up.
+    // This is to avoid `over-quoting` during restart, when RT is not fully populated,
+    // result in mis-calculation of relevant records.
+    pub fn cleanup_unrelevant_records(&mut self) {
+        let accumulated_records = self.records.len();
+        if accumulated_records < 6 * MAX_RECORDS_COUNT / 10 {
+            return;
+        }
+
+        let responsible_range = if let Some(range) = self.responsible_distance_range {
+            range
+        } else {
+            return;
+        };
+
+        let mut removed_keys = Vec::new();
+        self.records.retain(|key, _val| {
+            let kbucket_key = KBucketKey::new(key.to_vec());
+            let is_in_range =
+                responsible_range >= self.local_key.distance(&kbucket_key).ilog2().unwrap_or(0);
+            if !is_in_range {
+                removed_keys.push(key.clone());
+            }
+            is_in_range
+        });
```

Comment:
I think it is better to have the remove functionality at one place. `self.remove(key)` also removes it from `self.record` + cleans up from FS.
We might not gain much time saving from `retain` as well as `self.remove()` interacts with fs anyway.

Reply:
the reason of remove those keys separately here, is because the `remove` function will try to re-calculate furthest_key, if the to_be_removed key is the current furthest_key.
which, could cause duplicated costly calculation if call remove function individually.

meanwhile, `remove a non-existing entry` is not that costive.

Reply:
Cool, that makes sense!